regression_tests/regression_workflow_test.go

```diff
@@ -0,0 +1,332 @@
+		}{
+			{
+				name:  fmt.Sprintf("%s should add xPub record %s for %s", spvWalletPG.admin.paymail, spvWalletPG.user.xPub, spvWalletPG.user.paymail),
+				admin: spvWalletPG.admin,
+				user:  spvWalletPG.user,
+			},
+			{
+				name:  fmt.Sprintf("%s should add xPub record %s for %s", spvWalletPG.admin.paymail, spvWalletSL.user.xPub, spvWalletSL.user.paymail),
+				admin: spvWalletSL.admin,
+				user:  spvWalletSL.user,
+			},
+		}
+		for _, tc := range tests {
+			t.Run(tc.name, func(t *testing.T) {
+				// when:
+				xPub, err := tc.admin.client.CreateXPub(ctx, &commands.CreateUserXpub{XPub: tc.user.xPub})
+				// then:
+				if err != nil {
+					t.Errorf("xPub record %s wasn't created successfully for %s by %s. Got error: %v", tc.user.xPub, tc.user.paymail, tc.admin.paymail, err)
+				}
+
+				if xPub == nil {
+					t.Errorf("Expected to get non-nil xPub response after sending creation request by %s.", tc.admin.paymail)
+				}
+
+				if xPub != nil && err == nil {
+					t.Logf("xPub record %s was created successfully for %s by %s", tc.user.xPub, tc.user.paymail, tc.admin.paymail)
+				}
+			})
+		}
+	})
+
+	t.Run("Step 4: The SPV Wallet admin clients attempt to add paymail record of the user from the same env by making request to their SPV Wallet API instance", func(t *testing.T) {
+		tests := []struct {
+			name  string
+			user  *user
+			admin *admin
+		}{
+			{
+				name:  fmt.Sprintf("%s should add paymail record %s for the user %s", spvWalletPG.admin.paymail, spvWalletPG.user.paymail, spvWalletPG.user.alias),
+				admin: spvWalletPG.admin,
+				user:  spvWalletPG.user,
+			},
+			{
+				name:  fmt.Sprintf("%s should add paymail record %s for the user %s", spvWalletPG.admin.paymail, spvWalletSL.user.paymail, spvWalletSL.user.alias),
+				admin: spvWalletSL.admin,
+				user:  spvWalletSL.user,
+			},
+		}
+		for _, tc := range tests {
+			t.Run(tc.name, func(t *testing.T) {
+				// when:
+				paymail, err := tc.admin.client.CreatePaymail(ctx, &commands.CreatePaymail{
+					Key:        tc.user.xPub,
+					Address:    tc.user.paymail,
+					PublicName: "Regression tests",
+				})
+
+				// then:
+				if err != nil {
+					t.Errorf("Paymail record %s wasn't created successfully for %s by %s. Got error: %v", tc.user.paymail, tc.user.alias, tc.admin.paymail, err)
+				}
+
+				if paymail == nil {
+					t.Errorf("Expected to get non-nil paymail addresss response after sending creation request by %s.", tc.admin.paymail)
+				}
+
+				if err == nil && paymail != nil {
+					t.Logf("Paymail record %s was created successfully for %s by %s.", tc.user.paymail, tc.user.alias, tc.admin.paymail)
+				}
+			})
+		}
+	})
+
+	t.Run("Step 5: The leaders from one env attempts to transfer funds to users from another env using the appropriate SPV Wallet API instance", func(t *testing.T) {
+		tests := []struct {
+			name   string
+			leader *user
+			user   *user
+			funds  uint64
+		}{
+			{
+				leader: spvWalletPG.leader,
+				user:   spvWalletSL.user,
```

Comment:
hmm... we're not testing one of the scenerios now.
We should have:
step 5: leader -> user in the same env
step 6: user -> user in another env

This way we're testing:
step 5: internal transfers
step 6: external transfers